runtime/runtime/src/state_viewer/mod.rs

```diff
@@ -157,7 +157,20 @@ impl TrieViewer {
             });
         }
         // TODO(2076): Add proofs for the storage items.
-        Ok(ViewStateResult { values, proof: vec![] })
+        let trie = state_update.trie();
+        let root = state_update.get_root();
+
+        let proof = trie
+            .get_proof(&root, &query)?
+            .unwrap_or_else(|| vec![])
```

Comment:
@blasrodri : moving conversation from zulip here.  Do I understand correctly that you are getting a `None` here?

Reply:
In which case, we need to study `get_proof()` above to see under which conditions it can return `Ok(None)`.  Maybe try some `printf` debugging to see which case we are hitting and try to reason about why we are hitting that case.

Reply:
I think so. But I need to understand what is the expected result. 

```rs
    let result = trie_viewer.view_state(&state_update, &alice_account(), b"").unwrap();
    assert_eq!(result.proof, Vec::<String>::new());
```

when passing empty prefix, the proof is empty. Same with prefix being "xyz". Only when it is "test123" it gets the proof. I wonder whether this is correct, or should be modified. @akhi3030

Reply:
@blasrodri: I am stumped too!  I don't see any obvious issues with the implementation.  We probably need some other folks from NEAR to weigh in on this.  I suggest that you post in #general channel on zulip.